src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java

```diff
@@ -79,11 +79,6 @@ protected DetailAST getRightCurly() {
         return getMainAst().findFirstToken(TokenTypes.RCURLY);
     }
 
-    @Override
-    protected boolean canChildrenBeNested() {
```

Comment:
Changes discussed here - #8038 (comment)
On removing this method I get correct indentation results and no errors with #8038 (comment) is encountered.

This method has somehow no results on nesting of arrayInit children and no changes are encountered on removing or keeping this method for any arrayInit test cases inside IndentationcheckTest.java. Infact generated snapshot checks also don't have any effect on nested array init in any possible way.
Also on the file mentioned at comment link above the generated snapshot is not producing such errors.
keeping this method generate pitest failures, so I had to remove that.
Since it now violating with many online ci test cases, I think I have to restore this method and test further to remove pitest failures.